lib/scsocket.js

```diff
@@ -677,47 +677,51 @@ SCSocket.prototype._handleEventAckTimeout = function (eventObject, eventNode) {
   }
 };
 
-SCSocket.prototype._emit = function (event, data, callback) {
+SCSocket.prototype._emit = function (event, data) {
   var self = this;
 
-  if (this.state == this.CLOSED) {
-    this.connect();
-  }
-  var eventObject = {
-    event: event,
-    callback: callback
-  };
+  return new Promise(function(resolve,reject) {
```

Comment:
Since we can use the latest technology, we should use the fat arrow `() => {}` syntax instead of a `function`, that way we can just use `this` instead of `self` inside it.